doc/bgp_pic/bgp_pic.md

```diff
@@ -0,0 +1,200 @@
+<!-- omit in toc -->
+# BGP PIC HLD
+<!-- omit in toc -->
+### Revision
+| Rev |     Date    |       Author           | Change Description                |
+|:---:|:-----------:|:----------------------:|-----------------------------------|
+| 0.1 | Oct  8 2023 |   Eddie Ruan / lingyu Zhang   | Initial Draft              |
+
+<!-- omit in toc -->
+## Table of Content
+- [Goal \& Scope](#goal--scope)
+
+## Goal and Scope
+BGP PIC, as detailed in the RFC available at https://datatracker.ietf.org/doc/draft-ietf-rtgwg-bgp-pic/, addresses the enhancement of BGP route convergence. This document outlines a method to arrange forwarding structures that can lead to improved BGP route convergence. BGP PIC offers two primary enhancements:
+
+1. It prevents BGP load balancing updates from being triggered by IGP load balancing updates. This issue, which is discussed in the SONiC Routing Working Group (https://lists.sonicfoundation.dev/g/sonic-wg-routing/files/SRv6%20use%20case%20-%20Routing%20WG.pptx), can be effectively resolved using BGP PIC.
+
+<figure align=center>
+    <img src="images/srv6_igp2bgp.jpg" >
+    <figcaption>Figure 1. Alibaba issue Underlay routes flap affecting Overlay SRv6 routes <figcaption>
+</figure> 
+Note: we only handle VPN overlay routes via BGP PIC. For global table's recursive routes handling, it would be handled via a seperate HLD and done by Accton team.
+
+2. We aim to achieve fast convergence in the event of a hardware forwarding failure related to a remote BGP PE becoming unreachable. Convergence in the slow path forwarding mode is not a priority.
+
```

Comment:
Yes. It is BGP PIC core. 

For IGP nexthop changes, currently, I don't plan to handle it in PIC since SRv6 VPN doesn't have IGP label context. Also the benefit for adding IGP in PIC is much less than BGP. I would wait for use cases. But current approach could be applied to IGP PIC as well. Therefore IGP nexthop update, I would keep them as is for now.

Reply:
I just rephrase my wording "IGP nexthop PE changes". I don't mean the PE nexthop IP changes.
The remote PE nexthop IP does not change.
```
PE1 ------ NHA ----- {IGP core} ---- NH4 ----- PE2
|                                       |
|_______NHB ------ {IGP core} ---- NH3 ----- PE3
```
I looked at the SRv6usecase-RoutingWG.pptx document.
- IGP reachability goes from "via nexthop A and B" to "via nexthop A" : covered by PIC NHG
- what happens if "via nexthop B" comes again ? do you let the BGP handle it ?
- what happens if "via nexthop C" is appended ? do you let the BGP handle it ?

Adding nexthops in IGP is handled by slow path, while removing nexthops from IGP is handled by PIC NHG ?
""

Reply:
Since in our case, underlay is handled via eBGP and overlay is handled via MP-BGP, let's use IGP for eBGP and BGP for overlay's MP-BGP to make discussion easier.

IGP convergence could be done either via IGP (remote IGP link change case) or via zebra detecting local link fail. 

BGP NHG would not need to be updated if its reachability doesn't get changed in IGP.

Reply:
ok for the wording: IGP for eBGP and BGP for overlay.
"handled by slow path" => I understand that some cases will not be covered by modifying PIC NHG in zebra. Instead, BGP will detect the change, and will update the routes accordingly.
The use case talks about a path that disappears.
But what about the other use cases I described, just to be clear:
 - nexthop B comes again?
 - a nexthop C appears for the first time ?
Are they handled by zebra events that directly modify the BGPNHG pic context?

Reply:
"I understand that some cases will not be covered by modifying PIC NHG in zebra. Instead, BGP will detect the change, and will update the routes accordingly." Agree. The main goal for PIC is to reduce hardware traffic outage window. 

In your example, these are IGP NH update cases. If nexthop B comes again, or a nexthop C comes up, they are the same case. It would be handled via IGP convergence. For this case, the main goal is to avoid BGP routes to be refreshed again since BGP NH's reachability is not changed.